### ISSUE TITLE
caffe2：Some confusion about affinechannel

### ISSUE THREAD

**Issue by wytcsuch** [pytorch/pytorch#15336]
Hi,I have some confusions about affinechannel.I don't know if I understand it correctly.It seems that the function of the affinechannel is to add batchnorm layer and scale layer.
And I use the affinechannel to construct densenet model in detectron.The code is

   ```
    cur = model.AffineChannel(blob_in, prefix + '_x1_bn', dim = dim_in, inplace=True)
    cur = model.Relu(cur, cur)
    cur = model.Conv(cur, prefix + '_x1', dim_in, 4*k, kernel= 1, stride=1, no_bias=1)  #bias_term = fasle
```
the blob_in is pool3，when the inpalce = True，The following problems will arise

```
Gradient operator needs output "gpu_0/pool3" at version 0, but currently we have version 2.

DEBUG HELP:
Maybe you use same output blob twice for different ops?
== Version history of blob [gpu_0/pool3]
Version (out) 0 <-- input: "gpu_0/conv3_blk"
output: "gpu_0/pool3"
name: ""
type: "AveragePool"
arg {
  name: "kernel"
  i: 2
}
arg {
  name: "stride"
  i: 2
}
arg {
  name: "cudnn_exhaustive_search"
  i: 0
}
arg {
  name: "order"
  s: "NCHW"
}
device_option {
  device_type: 1
  cuda_gpu_id: 0
}
engine: "CUDNN"

Version (out) 1 <-- input: "gpu_0/pool3"
input: "gpu_0/conv4_1_x1_bn_s"
input: "gpu_0/conv4_1_x1_bn_b"
output: "gpu_0/pool3"
name: ""
type: "AffineChannel"
device_option {
  device_type: 1
  cuda_gpu_id: 0
}

Version (out) 2 <-- input: "gpu_0/pool3"
output: "gpu_0/pool3"
name: ""
type: "Relu"
arg {
  name: "cudnn_exhaustive_search"
  i: 0
}
arg {
  name: "order"
  s: "NCHW"
}
device_option {
  device_type: 1
  cuda_gpu_id: 0
}
engine: "CUDNN"
```
so I wonder to know the effect of the affinechannel,thank you very much.
